server/game/Card.js

```diff
@@ -137,14 +84,52 @@ class Card extends EffectSource {
     }
 
     get name() {
-        const copyEffect = this.mostRecentEffect('copyCharacter');
+        const copyEffect = this.mostRecentEffect('copyCard');
         return copyEffect ? copyEffect.printedName : this.printedName;
     }
 
     get type() {
         return this.mostRecentEffect('changeType') || this.printedType;
     }
 
+    get actions() {
+        let actions = this.abilities.actions;
+        if(this.anyEffect('copyCard')) {
+            let mostRecentEffect = _.last(this.effects.filter(effect => effect.type === 'copyCard'));
+            actions = mostRecentEffect.value.getActions(this);
+        }
+
+        let effectActions = this.getEffects('gainAbility').filter(ability => ability.abilityType === 'action');
+        return actions.concat(effectActions);
+    }
+
+    get reactions() {
+        const TriggeredAbilityTypes = ['interrupt', 'reaction', 'constant'];
+        let reactions = this.abilities.reactions;
+        if(this.anyEffect('copyCard')) {
+            let mostRecentEffect = _.last(this.effects.filter(effect => effect.type === 'copyCard'));
+            reactions = mostRecentEffect.value.getReactions(this);
+        }
+
+        let effectReactions = this.getEffects('gainAbility').filter(ability => TriggeredAbilityTypes.includes(ability.abilityType));
+        return reactions.concat(effectReactions);
+    }
+
+    get persistentEffects() {
+        let gainedPersistentEffects = this.getEffects('gainAbility').filter(ability => ability.abilityType === 'persistentEffect');
+        if(this.anyEffect('copyCard')) {
+            let mostRecentEffect = _.last(this.effects.filter(effect => effect.type === 'copyCard'));
+            return gainedPersistentEffects.concat(mostRecentEffect.value.getPersistentEffects());
+        }
+
+        return this.isBlank() ? gainedPersistentEffects : gainedPersistentEffects.concat(this.abilities.persistentEffects);
```

Comment:
How does this react with the ruling that upgrades that say 'gains: blah blah' blanks the gained stuff too?